lib/engine/spender.rb

```diff
@@ -4,32 +4,35 @@
 
 module Engine
   module Spender
-    attr_accessor :cash
-
-    def debt
-      @debt || 0
+    def spender
+      @spender || self
     end
 
-    def permanent_debt
-      @permanent_debt || 0
+    def spender=(entity)
+      raise GameError, "Cannot set (#{name}).spender to a non-<Engine::Spender>" unless entity.spender?
+
+      @spender = entity
     end
 
-    def check_cash(amount, borrow_from: nil)
-      available = @cash + (borrow_from ? borrow_from.cash : 0)
-      raise GameError, "#{name} has #{@cash} and cannot spend #{amount}" if (available - amount).negative?
+    def cash
+      spender == self ? @cash : spender.cash
     end
 
-    def check_positive(amount)
-      raise GameError, "#{amount} is not valid to spend" unless amount.positive?
+    def debt
+      spender == self ? (@debt || 0) : spender.debt
```

Comment:
Is there a reason not to have @debt protected too?

Reply:
I do have `debt=` protected like `cash=`. The public `debt` and `cash` methods here are wrappers similar to `attr_reader`